pac4j-core/src/main/java/org/pac4j/core/config/ConfigInstance.java

```diff
@@ -0,0 +1,48 @@
+/*
+  Copyright 2012 - 2015 pac4j organization
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.pac4j.core.config;
+
+import org.pac4j.core.exception.TechnicalException;
+
+/**
+ * One instance of the configuration. Useful in implementations where, for any reason,
+ * the configuration must be shared across several elements.
+ *
+ * @author Jerome Leleu
+ * @since 1.8.0
+ */
+public class ConfigInstance {
+
+    protected static Config config = new Config();
+
+    public synchronized static void build(final String factoryName) {
```

Comment:
No, generally, you only have one factory and one config instance per application (or none). Though, from a design POV, it would be better to have a class in charge of building configuration, distinct from the one that stores the configuration.